### DISCUSSION TITLE
VPCs

### DISCUSSION THREAD

**Discussion by anne8rete** [LAPKB/Pmetrics#67] [Q&A] [answered]
Hi! 

I'm not getting the VPCs to work, I have somewhat scattered data (clinical study/TDM data) and more often than not I keep getting this error: 

Error in chol.default(x) : 
  the leading minor of order 4 is not positive definite

Is there a way around this or more likely a data issue? This error has come up with different datasets and different models. 

I've tried running different options (TAD option seems to get furthest): 

valid_53 <- run53$validate(limits = NA) 
valid_53 <- run53$validate(limits = NA, TAD=T)
valid_53 <- run53$validate(limits = c(0,3)) 

The workshop models/data files do create nice pcVPC plots and NPDE :) 

Thanks!

Anne

**Comment by Siel** [comment]
Hi @anne8rete, coul you please share with me an example model/data for me to reproduce your error? it can be here or via e-mail: juliandavid347@gmail.com.

Thanks!

**Comment by mnneely** [comment]
This error arises generally when a model is "fragile", either with few support points or sparse data relative to the number of parameters. The covariance matrix is not positive definite. Normally, Pmetrics warns about this and offers to "tweak" the matrix as a work around. Did that not happen in this case?

**Reply by anne8rete** [reply]
No, it seems to happen after the simulation, when it is trying to create the files. I haven't gotten the matrix error/warning.
@Siel has a dummy dataset from me to check the error. 
When I try different options for limits, I tend to also get this error:
Error in `$<-.data.frame`(`*tmp*`, "PRED_bin", value = c(6.408734, 3.125139,  : 
  replacement has 1130 rows, data has 1132

But then after it still goes to the chol.default one.

**Reply by mnneely** [reply]
Try the new Version2 branch. I was working on the function for another error with multiple outputs, and made NPDE more robust by trying a different decorrelation method if Cholesky fails. Maybe that will work for you too?

**Reply by anne8rete** [reply]
Still getting the same error, it happens after the simulation though, at this stage:

Computing the npde for subject  11007 
Computing the npde for subject  11008 
Computing the npde for subject  11010 
Error in chol.default(x) : 
  the leading minor of order 4 is not positive definite

**Comment by Siel** [comment]
I was able to reproduce the error and I'm working to fix it at the moment

**Comment by mnneely** [comment, marked as answer]
I believe we confirmed that this error is fixed in current beta Version2 branch.

**Reply by Larubei** [reply]
Hello @mnneely @Siel 
I have the same issue with my personal data and model files (although PMvalid() command with example files works)
This sends me this message after the sim files are created :

_Error in `$<-.data.frame`(`*tmp*`, "PRED_bin", value = c(92.25062, 92.25062,  : 
  le tableau de remplacement a 13 lignes, le tableau remplacé en a 11_

I have the Pmetrics 2.0.2 version

Thanks a lot for your help
